external/yolo_detector/yolo/__mess.py

```
@@ -0,0 +1,288 @@
+"""
+
+The code of this file is only a copy of Jupyter Notebook code snippet.
+It's a MESS
+
+"""
+# -----------------------------------------------------------------------------------------------------
+import torch
+from collections import OrderedDict
+
+a = m2.state_dict()
+b = c2
+n = []
+for k1, k2 in zip(a, b):
+    n.append((k1, b[k2]))
+torch.save(OrderedDict(n), "xxx.pth")
+# -----------------------------------------------------------------------------------------------------
+import torch
+import yolo
+
+def _range(*args):
+    return list(range(*args))
+
+def bottle2concat(s, l):
+    tmp = [range(s,s+6),range(s+19,s+19+l*12),range(s+7,s+5,-1),range(s+14,s+19),range(s+8,s+14)]
+    order = [o for r in tmp for o in r]
+    return order
+
+ckpt = torch.load("/data/nextcloud/dbc2017/files/jupyter/data/ckpt/converted_yolov5s.pt")
+msd = ckpt["model"]
+del msd["model.27.anchors"], msd["model.27.anchor_grid"]
+
+model = yolo.YOLOv5(80)
+tgt = model.state_dict()
+
+stages = [12, 49, 110]
+layers = [1, 3, 3]
+orders = []
+for s, l in zip(stages, layers):
+    cur = _range(s - 6, s) + bottle2concat(s, l)
+    orders.extend(cur)
+    
+backbone = _range(0, 6) + orders + _range(165, 183)
+
+
+stages = [183, 220, 257, 296, 335]
+layers = [1, 1, 1, 1, 1]
+csp = [bottle2concat(s, l) for s, l in zip(stages, layers)]
+conv = [_range(x, x + 6) for x in [214, 251, 290, 329]]
+pred = [_range(x, x + 2) for x in [288, 327, 366]]
+
+new = [csp[2],csp[1],csp[0],conv[1],conv[0],conv[2],conv[3],csp[3],csp[4],pred[0],pred[1],pred[2]]
+panet = []
+for n in new:
+    panet.extend(n)
+    
+total = backbone + panet
+#print(len(panet), len(set(panet)), max(panet) - min(panet) + 1)
+#print(panet)
+norm = {total.index(i): v for i, v in enumerate(msd.values())}
+total_new = {k: norm[i] for i, k in enumerate(tgt.keys())}
+
+model.load_state_dict(total_new)
```

Something i spotted with GPT assistance, worth checking it out:

This line blindly loads reordered weights into a new model without any strict=False or verification:

If total_new is incomplete, this may silently skip weights.
If shapes mismatch, it will crash without debugging info.
Add verification:
```python
missing, unexpected = model.load_state_dict(total_new, strict=False)
print("Missing keys:", missing)
print("Unexpected keys:", unexpected)
```